app/views/todos/_todo_modal.html.haml

```diff
@@ -0,0 +1,20 @@
+#todomodal.modal.fade{"aria-hidden" => "true", "aria-labelledby" => "exampleModalLabel", :role => "dialog", :tabindex => "-1"}
+  .modal-dialog{:role => "document"}
+    .modal-content
+      .modal-header
+        %h5#exampleModalLabel.modal-title New ToDo
+        %button.close{"aria-label" => "Close", "data-dismiss" => "modal", :type => "button"}
+          %span{"aria-hidden" => "true"} ×
+      .modal-body
+        = form_with model: @todo do |form|
+          %div
+            = form.label :title
+            %br/
+            = form.text_field :title
+          %div
+            = form.label :text
+            %br/
+            = form.text_area :text
+          %div
+            = form.submit
+            
```

Comment:
As a suggestion, you may want to consider the usage of hashes having atoms instead of strings (when possible) as well as using single quotes when string interpolation is not required.
For example:
`role: 'dialog'` instead of `:role => "dialog"`